pkg/agent/openflow/pipeline.go

```diff
@@ -2292,26 +2291,25 @@ func (f *featureService) serviceLearnFlow(groupID binding.GroupIDType,
 	svcPort uint16,
 	protocol binding.Protocol,
 	affinityTimeout uint16,
-	nodeLocalExternal bool,
-	svcType v1.ServiceType) binding.Flow {
+	externallyAccessible bool,
+	isNodePort bool) binding.Flow {
 	// Using unique cookie ID here to avoid learned flow cascade deletion.
 	cookieID := f.cookieAllocator.RequestWithObjectID(f.category, uint32(groupID)).Raw()
-	var flowBuilder binding.FlowBuilder
-	if svcType == v1.ServiceTypeNodePort {
-		unionVal := (ToNodePortAddressRegMark.GetValue() << ServiceEPStateField.GetRange().Length()) + EpToLearnRegMark.GetValue()
-		flowBuilder = ServiceLBTable.ofTable.BuildFlow(priorityLow).
-			Cookie(cookieID).
-			MatchProtocol(protocol).
-			MatchRegFieldWithValue(NodePortUnionField, unionVal).
-			MatchDstPort(svcPort, nil)
+	flowBuilder := ServiceLBTable.ofTable.BuildFlow(priorityLow).
+		Cookie(cookieID).
+		MatchProtocol(protocol).
+		MatchDstPort(svcPort, nil)
+
+	// EpToLearnRegMark is required to match the packets that have done Endpoint selection.
+	regMarksToMatch := []*binding.RegMark{EpToLearnRegMark}
+	// If Service type is NodePort, ToNodePortAddressRegMark must be used to match packets, otherwise Service IP must be
```

Comment:
must be -> are

Reply:
Please review #4866, I'll updated this in it, thanks.